service/frontend/workflow_handler.go

```diff
@@ -235,6 +240,26 @@ func (wh *WorkflowHandler) Start() {
 			wh.healthInterceptor.SetHealthy(true)
 			wh.logger.Info("Frontend is now healthy")
 		}()
+
+		wh.namespaceRegistry.RegisterStateChangeCallback(wh, func(ns *namespace.Namespace, deletedFromDb bool) {
+			if deletedFromDb {
+				return
+			}
+
+			if ns.IsGlobalNamespace() &&
+				ns.ReplicationPolicy() == namespace.ReplicationPolicyMultiCluster &&
+				ns.ActiveClusterName() != wh.clusterMetadata.GetCurrentClusterName() {
+				pollers, ok := wh.outstandingPollers.Get(ns.ID().String())
+				if ok {
+					for _, key := range pollers.KeySet() {
```

Comment:
hmm.. since you need to get both keys and values, maybe the SyncMap interface you want is actually `GetAndClear`, that returns the current map and then resets it to an empty one? then you can just through and clear them all without additional locking.

Reply:
change to popAll